src/docfx/lib/collections/WorkQueue.cs

```diff
@@ -54,13 +48,18 @@ public void Enqueue(T item)
                 return;
             }
 
+            if (_drainTcs.Task.IsCompleted)
+            {
+                throw new InvalidOperationException("Cannot enqueue new items after the queue is drained");
+            }
+
             _queue.Enqueue(item);
 
             Interlocked.Increment(ref _totalCount);
             Interlocked.Increment(ref _remainingCount);
         }
 
-        public Task Drain(Action<int, int> progress = null)
+        public Task Drain(Func<T, Task> run, Action<int, int> progress = null)
```

Comment:
When I call Drain more than one time, they will execute different actions to the same queue, which will brings unexpected result.

Unless we need make sure the Drain can be called only once here.